lib/improver/blending/weights.py

```diff
@@ -238,12 +242,565 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+                for setting up the interpolation and create the new weights
+                cube.
+
+        Returns:
+            new_weights_cube (iris.cube.Cube):
+                Cube containing the output from the interpolation. This
+                has been renamed using the self.weights_coord_name but
+                otherwise matches the input cube.
+        """
+        cube_slices = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over(self.config_coord_name):
+            new_weights_cube = (
+                self._interpolate_to_create_weights(cube_slice))
+            cube_slices.append(new_weights_cube)
+
+        # Normalise the weights.
+        new_weights_cube = cube_slices.merge_cube()
+        axis = new_weights_cube.coord_dims(self.config_coord_name)
+        new_weights_cube.data = (
+            WeightsUtilities.normalise_weights(
+                new_weights_cube.data, axis=axis))
+        return new_weights_cube
+
+
+class ChooseWeightsLinearFromCube(object):
+    """Calculate weights when providing an input cube containing a coordinate
+       over which the weights will be calculated e.g. forecast_period.
+       As an input cube is supplied, the weights can potentially vary
+       spatially.
+       The weights will be applied to a separate coordinate e.g. model_id, so
+       that the output will be a cube with different weights for each model_id
+       as the forecast period varies."""
+
+    def __init__(self, weighting_coord_name, config_coord_name):
+        """Set up for calculating linear weights from a cube of weights, where
+        for example, the weights vary along the forecast_period coordinate.
+
+        Args:
+            weighting_coord_name (str):
+                Name of the coordinate along which the weights will be
+                calculated. For example, if the intention is to provide
+                weights along the forecast_period coordinate within the
+                configuration dictionary, which can be linearly interpolated
+                to create all required weights, then this argument would be
+                "forecast_period".
+                This must be included within the configuration dictionary.
+            config_coord_name (str):
+                Name of the coordinate used for configuration. For example,
+                if the intention is to create weights for different models,
+                then a model_configuration may appropriate.
+
+        """
+        self.weighting_coord_name = weighting_coord_name
+        self.config_coord_name = config_coord_name
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        msg = ("<ChooseWeightsLinearFromCube weighting_coord_name = {}, "
+               "config_coord_name = {}>".format(
+                   self.weighting_coord_name, self.config_coord_name))
+        return msg
+
+    def _check_weight_cubes(self, cube, weights_cubes):
```

Comment:
"weight" is plural everywhere except in this function title (I think). Please change to "weights" to be consistent.

Reply:
Done.